langserver/symbol.go

```diff
@@ -71,11 +76,27 @@ func queryJoin(s, e string) string {
 // ParseQuery parses a user's raw query string and returns a
 // structured representation of the query.
 func ParseQuery(q string) (qu Query) {
-	// All queries are case insensitive.
-	q = strings.ToLower(q)
+	qu = doParseQuery(q)
+	if !qu.CaseSensitive {
+		q = strings.ToLower(qu.String())
+		qu = doParseQuery(q)
+	}
+	return qu
+}
 
+func doParseQuery(q string) (qu Query) {
 	// Split the query into space-delimited fields.
 	for _, field := range strings.Fields(q) {
+		// Check if the field is like `casesensitive:true`.
+		if strings.HasPrefix(field, "casesensitive:") {
+			if caseSensitive, err := strconv.ParseBool(strings.TrimPrefix(field, "casesensitive:")); err != nil {
```

Comment:
https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2232
wanted to inline there but couldn't - Non-name on left side.